examples/textual_inversion/textual_inversion.py

```diff
@@ -780,11 +786,24 @@ def main():
                 f" {args.validation_prompt}."
             )
             # create pipeline (note: unet and vae are loaded again in float32)
-            pipeline = DiffusionPipeline.from_pretrained(
-                args.pretrained_model_name_or_path,
-                text_encoder=accelerator.unwrap_model(text_encoder),
-                revision=args.revision,
-            )
+            if args.reuse_model_for_logging:
+                pipeline = DiffusionPipeline.from_pretrained(
```

Comment:
actually I think we should always use the new code because it'll always be better to re-use the UNet. Thanks a lot for finding this @isamu-isozaki :-) 

Could you maybe just replace:
```python
            pipeline = DiffusionPipeline.from_pretrained(
                args.pretrained_model_name_or_path,
                text_encoder=accelerator.unwrap_model(text_encoder),
                revision=args.revision,
            )
```

with:

```python
            pipeline = DiffusionPipeline.from_pretrained(
                    args.pretrained_model_name_or_path,
                    text_encoder=accelerator.unwrap_model(text_encoder),
                    tokenizer=tokenizer,
                    unet=unet,
                    vae=vae,
                    revision=args.revision,
                    torch_dtype=weight_dtype,
                )
```

and then we don't need a new arg variable

Reply:
Ah you're probably doing the changes in the pipeline because otherwise the "New" way of running inference fails with a dtype error.

Could you maybe solve this by wrapping everything into a:

```py
with torch.autocast("cuda"):
```

This way everything should be casted correctly and we don't have to do any changes to the pipeline code :-)